flume-ng-channels/flume-kafka-channel/src/main/java/org/apache/flume/channel/kafka/KafkaChannel.java

```diff
@@ -373,6 +372,7 @@ private void migrateOffsets() {
   }
 
   private void decommissionConsumerAndRecords(ConsumerAndRecords c) {
+    c.consumer.wakeup();
```

Comment:
Please ignore my prev comment I misread this change. Sorry for that. It is completely fine in the way you done it originally.